django/core/management/base.py

```diff
@@ -277,6 +279,18 @@ def create_parser(self, prog_name, subcommand, **kwargs):
         Create and return the ``ArgumentParser`` which will be used to
         parse the arguments to this command.
         """
+
+        def add_base_argument(parser, *args, **kwargs):
+            """
+            Call the parser's add_argument() method, suppressing the help text
+            according to BaseCommand.suppressed_base_arguments.
+            """
+            for arg in args:
```

Comment:
I think you still want the `-` logic I suggested in the optional argument case (when the first argument starts with `-`). If the first argument doesn't start with `-`, I believe you can restrict to looking at just the argument providing the single name.

Reply:
Thank you for suggestion, but I prefer the way it is now. 

Moreover I've tried to add positional argument with and w/o add_base_argument to `BaseCommand.create_parser`, but it fails because those arguments are always required, so the `AdminScriptTestCase` class would have to be refactored a bit to account for positional arguments or I'm missing sth there?

I would say that addressing positional arguments is for another task.
Regardless of that the loop logic works for either positional or optional arguments.

I hope that the patch in the current form is acceptable. I appreciate you help @cjerdonek.

Reply:
Hi @cjerdonek ,

I've committed additional assertion for positional argument named 'file' in with_suppress_command. I did have to copy the entire add_base_argument, as I don't see other way.